permit_jsligo/src/main.jsligo

```diff
@@ -0,0 +1,192 @@
+        let [l, e] = acc;
+        let [is_transfer_authorized, e] = Extension.transfer_presigned(e, t);
+        List.fold(
+          ( ([acc, dst] : [[FA2.Ledger.t, Extension.t], FA2.atomic_trans]) => {
+            let _ = FA2.Storage.assert_token_exist(s, dst.token_id);
+            // let _ = if (!is_transfer_authorized) {
+            //     FA2.Operators.assert_authorisation s.operators t.from_ dst.token_id
+            // };
+            if (!is_transfer_authorized) {
+                let _ = FA2.Operators.assert_authorisation(s.operators, t.from_, dst.token_id);
+
+                let [ledger, ext] = acc;
+                let ledger = FA2.Ledger.decrease_token_amount_for_user(ledger, t.from_, dst.token_id, dst.amount);
+                let ledger = FA2.Ledger.increase_token_amount_for_user(ledger, dst.to_, dst.token_id, dst.amount);
+                [ledger, ext]
+            } else {
+                let [ledger, ext] = acc;
+                let ledger = FA2.Ledger.decrease_token_amount_for_user(ledger, t.from_, dst.token_id, dst.amount);
+                let ledger = FA2.Ledger.increase_token_amount_for_user(ledger, dst.to_, dst.token_id, dst.amount);
+                [ledger, ext]
+            }
+            
+            // let [ledger, ext] = acc;
+            // let ledger = FA2.Ledger.decrease_token_amount_for_user(ledger, t.from_, dst.token_id, dst.amount);
+            // let ledger = FA2.Ledger.increase_token_amount_for_user(ledger, dst.to_, dst.token_id, dst.amount);
+            // [ledger, ext]
+          }), t.tx, [l, e]);
+    };
+    let [new_ledger, new_ext] = List.fold(make_transfer, t, [s.ledger, s.extension]);
+    return { ...s, ledger : new_ledger, extension : new_ext }
+};
+
+type parameter = [@layout:comb]
+    ["Transfer", FA2.transfer] |
+    ["Balance_of", FA2.balance_of] |
+    ["Update_operators", FA2.update_operators] |
+    ["Create_token", [create_token, address, nat]] |
+    ["Mint_token", list<mint_or_burn>] |
+    ["Burn_token", list<mint_or_burn>] |
+    ["Permit", list<permit_params>] |
+    ["SetExpiry", expiry_params] |
+    ["SetAdmin", address]
+
+let main = ([p,s]:[parameter, storage]): result => {
+    return match (p, {
+        Transfer:         (p: FA2.transfer) => [Constants.no_operation, transfer_permitted(p,s)],
+        Balance_of:       (p: FA2.balance_of) => FA2.balance_of(p, s),
+        Update_operators: (p: FA2.update_operators) => FA2.update_ops(p, s),
+        Create_token:     (p: [create_token, address, nat]) => create(p, s),
+        Mint_token:       (p: list<mint_or_burn>) => mint(p, s),
+        Burn_token:       (p: list<mint_or_burn>) => burn(p, s),
+        Permit:           (p: list<permit_params>) => [Constants.no_operation, permit(p, s)],
+        SetExpiry:        (p: expiry_params) => [Constants.no_operation, set_expiry(p, s)],
+        SetAdmin:         (p: address) => [Constants.no_operation, { ...s, extension : Extension.set_admin(s.extension, p) }]
```

Comment:
plz create a function like in my branch https://github.com/der-alter/permit-cameligo/blob/init/src/main.mligo#L145